Fails to run on windows due to USR1 signal handling

Cerbos expects `USR1` signal to temporarily switch the log level to `debug` level, however Windows doesn't have `syscall.SIGUSR1` which causes failure.

```
internal\observability\logging\logging.go:117:33: undefined: syscall.SIGUSR1
```

--- a/internal/observability/logging/logging.go
+++ b/internal/observability/logging/logging.go
@@ -6,9 +6,7 @@ package logging
 import (
 	"context"
 	"os"
-	"os/signal"
 	"strings"
-	"syscall"
 	"time"
 
 	grpc_zap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
@@ -91,37 +89,9 @@ func doInitLogging(ctx context.Context, level string) {
 			return lvl > zapcore.ErrorLevel
 		}))),
 	)
-
 	if minLogLevel > zap.DebugLevel {
 		handleUSR1Signal(ctx, minLogLevel, &atomicLevel)
 	}
-}
-
-// handleUSR1Signal temporarily sets the log level to debug when a SIGUSR1 signal is received.
-func handleUSR1Signal(ctx context.Context, originalLevel zapcore.Level, atomicLevel *zap.AtomicLevel) {
-	sigusr1 := make(chan os.Signal, 1)
-	signal.Notify(sigusr1, syscall.SIGUSR1)
-
-	go func() {
-		inProgress := false
-		doneChan := make(chan struct{}, 1)
-		extendChan := make(chan struct{}, 1)
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-sigusr1:
-				if inProgress {
-					extendChan <- struct{}{}
-				} else {
-					inProgress = true
-					go setLogLevelForDuration(ctx, doneChan, extendChan, originalLevel, atomicLevel)
-				}
-			case <-doneChan:
-				inProgress = false
-			}
-		}
-	}()
 }
 
 // setLogLevelForDuration temporarily sets the global log level to the given level for a period of time.

--- a/internal/observability/logging/signal.go
+++ b/internal/observability/logging/signal.go
@@ -1,0 +1,43 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !windows
+
+package logging
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// handleUSR1Signal temporarily sets the log level to debug when a SIGUSR1 signal is received.
+func handleUSR1Signal(ctx context.Context, originalLevel zapcore.Level, atomicLevel *zap.AtomicLevel) {
+	sigusr1 := make(chan os.Signal, 1)
+	signal.Notify(sigusr1, syscall.SIGUSR1)
+
+	go func() {
+		inProgress := false
+		doneChan := make(chan struct{}, 1)
+		extendChan := make(chan struct{}, 1)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-sigusr1:
+				if inProgress {
+					extendChan <- struct{}{}
+				} else {
+					inProgress = true
+					go setLogLevelForDuration(ctx, doneChan, extendChan, originalLevel, atomicLevel)
+				}
+			case <-doneChan:
+				inProgress = false
+			}
+		}
+	}()
+}

--- a/internal/observability/logging/signal_windows.go
+++ b/internal/observability/logging/signal_windows.go
@@ -1,0 +1,17 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+
+package logging
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func handleUSR1Signal(_ context.Context, _ zapcore.Level, _ *zap.AtomicLevel) {
+	return
+}

